pkg/network/cache/multi.go

```diff
@@ -33,21 +32,39 @@ func newMultiClient(addr network.AddressGroup, opts []client.Option) *multiClien
 
 // note: must be wrapped into mutex lock.
 func (x *multiClient) createForAddress(addr network.Address) clientcore.Client {
-	opts := append(x.opts, client.WithAddress(addr.HostAddr()))
+	var (
+		c       client.Client
+		prmInit client.PrmInit
+		prmDial client.PrmDial
+	)
 
+	prmDial.SetServerURI(addr.HostAddr())
 	if addr.TLSEnabled() {
-		opts = append(opts, client.WithTLSConfig(&tls.Config{}))
+		prmDial.SetTLSConfig(&tls.Config{})
 	}
 
-	c, err := client.New(opts...)
+	if x.opts.Key != nil {
+		prmInit.SetDefaultPrivateKey(*x.opts.Key)
+	}
+
+	if x.opts.DialTimeout != 0 {
```

Comment:
```suggestion
	if x.opts.DialTimeout > 0 {
```

Reply:
I think panic is ok here, node is likely misconfigured.